oprf-core/src/ddlog_equality_shamir.rs

```diff
@@ -98,33 +113,56 @@ impl DLogEqualitySession {
 
         // The following modular reduction in convert_base_to_scalar is required in rust to perform the scalar multiplications. Using all 254 bits of the base field in a double/add ladder would apply this reduction implicitly. We show in the docs of convert_base_to_scalar why this does not introduce a bias when applied to a uniform element of the base field.
         let e_ = crate::dlog_equality::convert_base_to_scalar(e);
-        DLogEqualityProofShare {
-            s: self.d + b * self.e + lagrange_coefficient * e_ * x_share,
-        }
+        DLogEqualityProofShare(self.d + b * self.e + lagrange_coefficient * e_ * x_share)
```

Comment:
This is a bit weakly typed, since the types for the shamir and non-shamir variants are the same, however, you cannot use them in the other protocol correctly. Do you think we should add a separate type here?